vscode-dotnet-runtime-library/src/Acquisition/GlobalInstallerResolver.ts

```diff
@@ -0,0 +1,351 @@
+/* tslint:disable:no-any */
+/* tslint:disable:only-arrow-functions */
+
+/**
+ * @remarks
+ * This is similar to the version resolver but accepts a wider range of inputs such as '6', '6.1', or '6.0.3xx' or '6.0.301'.
+ * It currently only is used for SDK Global acquistion to prevent breaking existing behaviors.
+ * Throws various errors in the event that a version is incorrectly formatted, the sdk server is unavailable, etc.
+ */
+export class GlobalInstallerResolver {
+    // The unparsed version into given to the API to request a version of the SDK.
+    // The word 'version' is 2nd in the name so that it's not auto-completed and mistaken for fullySpecifiedVersionRequested, which is what should be used.
+    private requestedVersion : string;
+
+    // The url for a the installer matching the machine os and arch of the system running the extension
+    private discoveredInstallerUrl : string;
+
+    // The properly resolved version that was requested in the fully-specified 3-part semver version of the .NET SDK.
+    private fullySpecifiedVersionRequested : string;
+
+    private releasesJsonErrorString = `The API hosting the dotnet releases.json is invalid or has changed and the extension needs to be updated. Invalid API URL: `;
+    private releasesJsonKey = 'releases';
+    private releasesSdksKey = 'sdks';
+    private releasesSdkRidKey = 'rid';
+    private releasesSdkFileKey = 'files';
+    private releasesSdkVersionKey = 'version';
+    private releasesSdkNameKey = 'name';
+    private releasesUrlKey = 'url';
+    private releasesLatestSdkKey = 'latest-sdk';
+
+    /**
+     * @remarks Do NOT set this unless you are testing.
+     * Written to allow mock data to be given to the resolver.
+     */
+    public customWebRequestWorker? : WebRequestWorker | null = null;
+
+    constructor(
+        private readonly extensionState: IExtensionState,
+        private readonly eventStream: IEventStream,
+        requestedVersion : string
+    )
+    {
+        this.requestedVersion = requestedVersion;
+        this.discoveredInstallerUrl = '';
+        this.fullySpecifiedVersionRequested = '';
+    }
+
+
+    /**
+     *
+     * @returns The url to the installer for the sdk that matches the machine os and architecture, as well as for the requestedVersion.
+     */
+    public async getInstallerUrl(): Promise<string>
+    {
+        if(this.discoveredInstallerUrl === '')
+        {
+            this.discoveredInstallerUrl = await this.routeRequestToProperVersionRequestType(this.requestedVersion);
+        }
+        return this.discoveredInstallerUrl;
+    }
+
+    /**
+     *
+     * @returns The fully specified version in a standardized format that was requested.
+     */
+    public async getFullVersion(): Promise<string>
+    {
+        if(this.fullySpecifiedVersionRequested === '')
+        {
+            this.discoveredInstallerUrl = await this.routeRequestToProperVersionRequestType(this.requestedVersion);
```

Comment:
nit: it looks like a bug that we're checking one variable and then setting another here. It might be more clear to either return the two vars from `routeRequestToProperVersionRequestType` as a tuple or just set both of them directly in the method.

Reply:
Good idea, I used the tuple method! I didn't know that was a possibility. It's much better to read that code now 😄